cookbooks/bcpc-hadoop/recipes/opentsdb.rb

```diff
@@ -34,5 +34,3 @@
   "hbase/_HOST@#{krb_realm}"
 node.force_default['bach_opentsdb']['hbase_region_princ'] =
   "hbase/_HOST@#{krb_realm}"
-
-include_recipe 'bach_opentsdb'
```

Comment:
this only makes the dependency implicit which is more confusing when debugging.

Reply:
This allows us to prevent accidental cyclical dependencies or unnecessary dependencies for Chef-BACH as we work to break bcpc up. As this is a wrapper recipe, we know that it needs the base cookbook and like most of our specific uses there is a role [BCPC-Hadoop-Head-OpenTSDB-Server](https://github.com/bloomberg/chef-bach/pull/1206/files#diff-db60cd2ab1f98652f0acda7d4127b893) that provides the ordering.

This was likely needed for simplifying rspec or chef-spec but is not necessary for the Chef12-Server work if you would still like me to revert the changes?

Reply:
we shouldn't rely on the role ordering for dependencies. it's brittle, unversioned and implicit. it's hard to debug if you have unsynced roles in production.

if we have cyclical dependencies, it will get detected with `berks install` and get the error at the build-level. nothin having this will just move implicit cyclical dependencies in runtime which bites is in production.

Reply:
This actually fixes at least one circular dependency with `bach_repository` which berks was not catching.

Reply:
the default solver doesn't detect circular dependencies. you would have to add `solver :ruby` (the molinillo gem) to be able to detect it.

Reply:
Restored to old behavior